test/boost/database_test.cc

```diff
@@ -682,6 +705,58 @@ SEASTAR_TEST_CASE(snapshot_skip_flush_works) {
     });
 }
 
+SEASTAR_TEST_CASE(snapshot_use_sstable_identifier_works) {
+#ifndef SCYLLA_ENABLE_ERROR_INJECTION
+        fmt::print("Skipping test as it depends on error injection. Please run in mode where it's enabled (debug,dev).\n");
+        return make_ready_future<>();
+#endif
+    sstring table_name = "cf";
+    utils::get_local_injector().enable("random_sstable_identifier", false);
+    return do_with_some_data({table_name}, [table_name] (cql_test_env& e) {
+        sstring tag = "test";
+        db::snapshot_options opts = {.use_sstable_identifier = true};
+        take_snapshot(e, "ks", table_name, tag, opts).get();
```

Comment:
noob question: IIUC, when sstables are initially created, their file names contain their real sstable IDs. The problematic thing happens on table migration, where sstable ID is preserved, but the file name is regenerated from scratch. Does this test somehow checks snapshot behavior when sstables in the data dir have different file names from their sstable IDs, or does it operate on the easy case of sstables with file names containing their real IDs? If not, this test would also pass without the change implemented in this PR, is that right?